src/identity/client.rs

```diff
@@ -96,6 +96,24 @@ impl FullId {
     pub fn public_id(&self) -> &PublicId {
         &self.public_id
     }
+
+    /// Constructs a `FullId` with a particular BLS secret key.
+    pub fn with_bls_key(bls_sk: BlsSecretKey) -> Self {
```

Comment:
To avoid code duplication, we should change `new_bls` to use this c'tor:
```rust
pub fn new_bls<T: CryptoRng + Rng>(rng: &mut T) -> Self {
    let bls_secret_key = rng.gen::<BlsSecretKey>();
    Self::with_bls_key(bls_secret_key)
}
```

Reply:
I needed to create a `ClientFullId` given a specific BLS key that was created during account creation.